scipy/stats/tests/test_resampling.py

```diff
@@ -732,124 +737,188 @@ def statistic_extradim(*args, axis):
+            monte_carlo_test(data, stats.norm.rvs, None)
 
         message = "`n_resamples` must be a positive integer."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat,
-                             n_resamples=-1000)
+            monte_carlo_test(data, stats.norm.rvs, stat, n_resamples=-1000)
 
         message = "`n_resamples` must be a positive integer."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat,
-                             n_resamples=1000.5)
+            monte_carlo_test(data, stats.norm.rvs, stat, n_resamples=1000.5)
 
         message = "`batch` must be a positive integer or None."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat, batch=-1000)
+            monte_carlo_test(data, stats.norm.rvs, stat, batch=-1000)
 
         message = "`batch` must be a positive integer or None."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat, batch=1000.5)
+            monte_carlo_test(data, stats.norm.rvs, stat, batch=1000.5)
 
         message = "`alternative` must be in..."
         with pytest.raises(ValueError, match=message):
-            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat,
-                             alternative='ekki')
+            monte_carlo_test(data, stats.norm.rvs, stat, alternative='ekki')
+
+        # *If* this raises a value error, make sure it has the intended message
+        message = "Signature inspection of statistic"
+        def rvs(size):
+            return xp.asarray(stats.norm.rvs(size=size))
+        try:
+            monte_carlo_test(data, rvs, xp.mean)
+        except ValueError as e:
+            assert str(e).startswith(message)
+
+    @array_api_compatible
+    def test_input_validation_xp(self, xp):
+        def non_vectorized_statistic(x):
+            return xp.mean(x)
+
+        message = "`statistic` must be vectorized..."
+        sample = xp.asarray([1., 2., 3.])
+        if is_numpy(xp):
+            monte_carlo_test(sample, stats.norm.rvs, non_vectorized_statistic)
+            return
+
+        with pytest.raises(ValueError, match=message):
+            monte_carlo_test(sample, stats.norm.rvs, non_vectorized_statistic)
+        with pytest.raises(ValueError, match=message):
+            monte_carlo_test(sample, stats.norm.rvs, xp.mean, vectorized=False)
 
     @pytest.mark.xslow
-    def test_batch(self):
+    @array_api_compatible
+    def test_batch(self, xp):
         # make sure that the `batch` parameter is respected by checking the
         # maximum batch size provided in calls to `statistic`
         rng = np.random.default_rng(23492340193)
-        x = rng.random(10)
+        x = xp.asarray(rng.standard_normal(size=10))
+
+        xp = array_namespace(x)
```

Comment:
I got lazy and didn't create a separate `xp_test`. We need array-api-compat version because the statistic uses `var` with `correction`. Hopefully that's OK.